### ISSUE TITLE
About Prefix vector in ablation study

### ISSUE THREAD

**Issue by paozhuanis1** [MinkyuKim26/Prefix_AAC_ICASSP2023#5]
Hello, I have another question when I want to replicate ablation study about the mapping network.

        if self.temporal_prefix_length > 0:
            # temporal_prefix_vector = self.temporal_mappingnetwork(temporal_feature).view(-1, self.temporal_prefix_length, self.gpt_embedding_size)
            temporal_prefix_vector = self.temporal_mappingnetwork(temporal_feature).view(-1, self.temporal_prefix_length, self.gpt_embedding_size).to(self.device)
        elif self.global_prefix_length + self.temporal_prefix_length == 0:
            temporal_feature = temporal_feature.permute(0, 2, 1, 3).contiguous()
            temporal_feature = torch.reshape(temporal_feature, (temporal_feature.size()[0], temporal_feature.size()[1], -1))
            temporal_prefix_vector = self.temporal_mappingnetwork(temporal_feature)

        if self.global_prefix_length > 0:
            global_prefix_vector = self.global_mappingnetwork(global_feature).view(-1, self.global_prefix_length, self.gpt_embedding_size)
            global_prefix_vector = global_prefix_vector.to("cuda:1")
        elif self.global_prefix_length + self.temporal_prefix_length == 0:
            global_prefix_vector = self.global_mappingnetwork(global_feature)
            global_prefix_vector = global_prefix_vector.view(global_feature.size()[0], 11, 768)  # [bsize, 11, 768]
            global_prefix_vector = global_prefix_vector.to("cuda:1")

            

        if self.temporal_prefix_length > 0 and self.global_prefix_length == 0:
            prefix_vectors = temporal_prefix_vector
        elif self.temporal_prefix_length == 0 and self.global_prefix_length > 0:
            prefix_vectors = global_prefix_vector
        else:
            prefix_vectors = torch.cat(
                (temporal_prefix_vector, global_prefix_vector), dim=1)

if I set self.global_prefix_length > 0 and self.temporal_prefix_length = 0,  does it means the method with a mapping network and global feature,vice versa.
And is I set self.global_prefix_length = 0 and self.temporal_prefix_length = 0, does it means the method without mapping networks, just two features which is fed into gpt2?

Thank you for your early reply!

**Comment by paozhuanis1** [comment]
I'm sorry for bothering you.
Also, I want to know how to use custom vocab size, I found the following section commented out.
` if self.Dataset == 'Clotho' and self.vocab_size != None :`
 `logits_for_clotho = self.language_header_only_for_clotho(out_hidden_states)`
`logits = torch.cat((logits, logits_for_clotho), dim=2)`
if I want to use custom one, I should set the vocab size to how much?

I would be appreciated for your early response!!!

**Comment by MinkyuKim26** [comment]
Hi, the answers are like the following.

Q. if I set self.global_prefix_length > 0 and self.temporal_prefix_length = 0, does it mean the method with a mapping network and global feature,vice versa.
A. Yes, if we set to use only global_prefix, then you only train&inference the mapping network for global feature.

Q. if I set self.global_prefix_length = 0 and self.temporal_prefix_length = 0, does it mean the method without mapping networks, just two features which is fed into gpt2?
A. If we set total prefix_length to 0, then linear layers(for matching the dimension) for both features are used. Therefore, the pseudo-code is like this.
global_feature = linear(global_feature) 
temporal_feature = linear(temporal_feature)
feature = torch.cat(temporal_feature, global_feature)
result = gpt2(feature)

Q. I want to know how to use custom vocab size
A. If you want to use the custom_vocabulary, then make the custom_vocab by using the 'make_vocabulary/make_dictionary.ipynb'.
After making the custom_vocabulary, then set the vocab size to len(custom_vocabulary)

If you have an additional question about this, please let me know.

Thank you.

**Comment by paozhuanis1** [comment]
Thank you for your answering!!!
But  I have confronted the problem when I set both temporal_prefix_size and global_prefix_size to 0.
The error message is ''the size of tensor a (88) must match the size of tensor b (31) at non-singleton dimension 3''
as following,
![QQ图片20240113103806](https://github.com/MinkyuKim26/Prefix_AAC_ICASSP2023/assets/86753693/4db049ae-74dd-43ce-a544-1712c9200aff)

I am wondering why it happened.
I'm sorry for bothering you again, and I would be appreciated for your early response!!!
Thank you very much!

**Comment by MinkyuKim26** [comment]
Hi,

Maybe checking the dataset code (Clotho of Audiocaps) may give a hint for solving this error.

If you have any questions, please let me know
